py/objstr.c

```diff
@@ -357,8 +357,7 @@ mp_obj_t mp_obj_str_binary_op(mp_binary_op_t op, mp_obj_t lhs_in, mp_obj_t rhs_i
         mp_obj_t *args = &rhs_in;
         size_t n_args = 1;
         mp_obj_t dict = MP_OBJ_NULL;
-        if (mp_obj_is_type(rhs_in, &mp_type_tuple)) {
-            // TODO: Support tuple subclasses?
+        if (mp_obj_is_subclass_fast(MP_OBJ_FROM_PTR(mp_obj_get_type(rhs_in)), MP_OBJ_FROM_PTR(&mp_type_tuple))) {
```

Comment:
This should use `mp_obj_is_tuple_compatible` so that it's also compatible with anonymous `attrtuple` objects.